packages/mobile/src/analytics/ValoraAnalytics.ts

```diff
@@ -0,0 +1,152 @@
+    isEmulator: await DeviceInfo.isEmulator(),
+    isTablet: DeviceInfo.isTablet(),
+  }
+}
+
+const SEGMENT_OPTIONS: analytics.Configuration = {
+  using: [Firebase],
+  flushAt: 20,
+  trackAttributionData: false,
+  debug: __DEV__,
+  trackAppLifecycleEvents: true,
+  recordScreenViews: true,
+  ios: {
+    trackAdvertising: false,
+    trackDeepLinks: true,
+  },
+}
+
+class ValoraAnalytics {
+  sessionId: string
+  userAddress: string | null | undefined
+  deviceInfo: object
+
+  constructor() {
+    this.sessionId = ''
+    this.userAddress = ''
+    this.deviceInfo = {}
+
+    if (!SEGMENT_API_KEY) {
+      Logger.debug(TAG, 'Segment API Key not present, likely due to environment. Skipping enabling')
+    }
+  }
+
+  init() {
+    Analytics.setup(SEGMENT_API_KEY, SEGMENT_OPTIONS).catch((error) =>
+      Logger.debug(TAG, `Segment setup error: ${error.message}\n`, error)
+    )
+    Logger.debug(TAG, 'Segment Analytics Integration initialized!')
+
+    getDeviceInfo()
+      .then((res) => {
+        this.deviceInfo = res
+        this.sessionId = Web3Utils.soliditySha3({
+          type: 'string',
+          value: res.SerialNumber || res.DeviceId,
+        }).slice(2)
+      })
+      .catch((err) => Logger.error(TAG, 'getDeviceInfo error', err))
+  }
+
+  isEnabled() {
+    // Remove __DEV__ here to test analytics in dev builds
+    return !__DEV__ && store.getState().app.analyticsEnabled
+  }
+
+  startSession(
+    eventName: typeof AppEvents.app_launched,
+    eventProperties: AnalyticsPropertiesList[AppEvents.app_launched]
+  ) {
+    this.track(eventName, {
+      deviceInfo: this.deviceInfo,
+      ...eventProperties,
+    })
+  }
+
+  setUserAddress(address?: string | null) {
+    this.userAddress = address
+  }
+
+  track<EventName extends keyof AnalyticsPropertiesList>(
+    ...args: undefined extends AnalyticsPropertiesList[EventName]
+      ? [EventName] | [EventName, AnalyticsPropertiesList[EventName]]
+      : [EventName, AnalyticsPropertiesList[EventName]]
+  ) {
+    const [eventName, eventProperties] = args
+
+    if (!this.isEnabled()) {
+      Logger.info(TAG, `Analytics is disabled, not tracking event ${eventName}`)
+      return
+    }
+
+    Logger.info(TAG, `Tracking event ${eventName}`, JSON.stringify(eventProperties))
+
+    if (!SEGMENT_API_KEY) {
+      return
+    }
+
+    const props: {} = {
+      timestamp: Date.now(),
+      sessionId: this.sessionId,
+      userAddress: this.userAddress,
+      ...eventProperties,
+    }
+
+    Analytics.track(eventName, props).catch((err) => {
+      Logger.error(TAG, `Failed to track event ${eventName}`, err)
+    })
+  }
+
+  page(page: string, eventProperties = {}) {
+    if (!SEGMENT_API_KEY) {
+      return
+    }
+
+    Analytics.screen(page, eventProperties).catch((err) => {
+      Logger.error(TAG, 'Error tracking page', err)
+    })
+  }
+}
+
+const ValoraAnalyticsInstance = new ValoraAnalytics()
+ValoraAnalyticsInstance.init()
```

Comment:
How about we call this from `index.js` or `App.tsx` for greater control of when it runs. By putting it here, it'll execute whenever this file happens to be first imported, and it'll block until init returns.

Reply:
Got it. Put it in `App` right before I call `startSession`. I considered folding `init` and `startSession` into one method but figured it's better to keep the concerns a little more separate.